### PR TITLE
Support passing arbitrary extra keys to fail_json_aws

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY

After getting caught out while debugging, support adding extra keys to fail_json_aws
Add some unit tests too for good measure

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

plugins/module_utils/core.py

##### ADDITIONAL INFORMATION

example of being caught out:
https://app.shippable.com/github/ansible-collections/amazon.aws/runs/666/20/tests

### CODE REVIEW

**Comment by tremble** [comment]
Will push changelog once tests have run.

**Comment by tremble** [comment]
Failure with the first run was the Python 3.9 unit tests failing due to the xtree change:
```
05:39 /usr/local/lib/python3.9/dist-packages/botocore/compat.py:149: in <module>
05:39     import xml.etree.cElementTree
05:39 E   ModuleNotFoundError: No module named 'xml.etree.cElementTree'
```

@jillr Did we have a fix for this?  I thought we were leaving the Python 3.9 tests disabled for now.

File: shippable.yml [hunk 1]
```diff
@@ -16,7 +16,8 @@ matrix:
     - env: T=units/3.6/1 A_REV=devel
     - env: T=units/3.7/1 A_REV=devel
     - env: T=units/3.8/1 A_REV=devel
-    - env: T=units/3.9/1 A_REV=devel
+# Until boto/botocore#2002 is fixed
```
Comment (jillr) [inline shippable.yml:19]: Sorry to be a pain but as you noted on irc upstream boto has merged the fix for this.  Testing this locally on 3.9 is passing now, can you back out this change?

**Comment by tremble** [comment]
@jillr I knew it had been merged, I hadn't expected a release that quickly!

File: shippable.yml (hunk 1 above)
Comment (tremble) [reply shippable.yml]: @jillr I knew it had been merged, I hadn't expected a release that quickly!

**Pull request merged**